src/tree.rs

```diff
@@ -51,6 +44,23 @@ pub struct SparseMerkleTree<H, V, S> {
     phantom: PhantomData<(H, V)>,
 }
 
+#[derive(Eq, PartialEq, Debug, Default, Hash, Clone)]
+struct H256OrdTree {
```

Comment:
Same thing as the above, can we define only one `H256Ord`, possibly in `h256.rs`?